router/core/request_context_fields.go

```diff
@@ -54,3 +54,81 @@ func NewDurationLogField(val time.Duration, attribute config.CustomAttribute) za
 	}
 	return zap.Skip()
 }
+
+func GetLogFieldFromCustomAttribute(field config.CustomAttribute, req RequestContext) zap.Field {
```

Comment:
Those methods shouldn't be public. You can also work with `reqContext := getRequestContext(ctx)` instead of doing a type assertion on the context.